sound/pci/hda/patch_hdmi.c

```diff
@@ -3026,6 +3087,27 @@ static int patch_i915_tgl_hdmi(struct hda_codec *codec)
 	return ret;
 }
 
+static int patch_i915_adlp_hdmi(struct hda_codec *codec)
+{
+	struct hdmi_spec *spec;
+	int res;
+
+	res = patch_i915_tgl_hdmi(codec);
+	if (!res) {
+		spec = codec->spec;
+		/*
+		 * For ADL-P, always enable standard HDA keep-alive (KAE).
+		 * As the codec can be powered down and keep-alive is
+		 * still sent, there is no need for SILENT_STREAM_I915
+		 * method on these platforms and the "enable_silent_stream"
+		 * module parameter can be ignored.
```

Comment:
nit-pick: for test purposes it could be interesting to disable this mode - e.g. to check if there's any difference...

Reply:
@plbossart Indeed. It's a bit of a shame "enable_silent_stream" is a boolean module parameter. For debug it would be also interesting to be able to force old-style silent streaming on new hw. But given this is a bit of a niche feature, I ended up not adding any new options and keeping this simple. If any odd bugs are seen, we can always extend the options (and then we'll have good reasons to do so).

Reply:
sounds good @kv2019i